the build system recognises armhf as armv7 (which it usually is..)
but for us that is wrong. it compiles anyway since clang is a cross compiler.
--- a/compiler-rt/cmake/base-config-ix.cmake
+++ b/compiler-rt/cmake/base-config-ix.cmake
@@ -230,9 +230,7 @@
       if(WIN32)
         test_target_arch(arm "" "" "")
       else()
-        test_target_arch(arm "" "-march=armv7-a" "-mfloat-abi=soft")
-        test_target_arch(armhf "" "-march=armv7-a" "-mfloat-abi=hard")
-        test_target_arch(armv6m "" "-march=armv6m" "-mfloat-abi=soft")
+        test_target_arch(armhf "" "-march=armv6zk" "-mfloat-abi=hard")
       endif()
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "avr")
       test_target_arch(avr "__AVR__" "--target=avr")
